### ISSUE TITLE
[explicit-function-return-type] `allowExpressions` false positives and incorrect documentation

### ISSUE THREAD

**Issue by gilbsgilbs** [typescript-eslint/typescript-eslint#387]
**Repro**

```JSON
{
  "rules": {
    "typescript/explicit-function-return-type": ["error", {allowExpressions: true}]
  }
}
```

```TS
node.addEventListener('click', () => {});
```

**Expected Result**

No error

**Actual Result**

Reported as missing return type

**Additional Info**

I can see at least two issues:

- As indicated [here](https://github.com/typescript-eslint/typescript-eslint/blob/bea6b92596942f551fedecf4d1fcc8551d103644/packages/eslint-plugin/docs/rules/explicit-function-return-type.md) the default value for `allowExpressions` should be `false` but [it's not](https://github.com/typescript-eslint/typescript-eslint/blob/cebcfe640ec56535bc565dabad6fb4b90db60afc/packages/eslint-plugin/src/rules/explicit-function-return-type.ts#L42)
- As the option is all borked, even when enabled (which is the actual default), the rule reports a ton of false positives which lets the user think that the option is disabled even when it's enabled (explicitly or implicitly). This is probably why we didn't had this reported before.

I investigated a bit and was able to figure out why tests are not failing: `filename` key is either not set or set to `'something.js'` in tests with `allowExpressions`. This tells eslint to skip all typescript rules somehow. I am easily able to reproduce in tests by setting the `filename` to `'test.ts'`.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `cebcfe64` |
| `@typescript-eslint/parser`        | `cebcfe64` |
| `TypeScript`                       | `3.3.3` |
| `ESLint`                           | `5.14.1` |
| `node`                             | `11.12.0` |
| `npm`                              | `6.9.0` |
